### ISSUE TITLE
CVE-2022-24130 crashes toyterm

### ISSUE THREAD

**Issue by ghost** [algon-320/toyterm#1]
As reported by @dankamongmen here: https://nvd.nist.gov/vuln/detail/CVE-2022-24130 , a large repeat count within a sixel sequence can cause a crash, either from buffer under/overflow or out of memory. toyterm git head dies with:

```
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 3', /home/autumn/t/toyterm/sixel/src/lib.rs:138:33
```

https://www.openwall.com/lists/oss-security/2022/01/30/3 has a nice test script:

```
#!/bin/bash
printf "\ePq"
printf "#%hhu;2;%hhu;%hhu;%hhu" 0x41 100 100 100
printf "#%hhu!%u@" 0x41 0x7fffffff
printf "#%hhu!%u@" 0x41 0x7fffffff
printf "\e\\"
```

Some additional discussion with other terminals is here: https://gitlab.com/klamonte/jexer/-/issues/105

**Comment by algon-320** [comment]
Thank you for your reporting!
I'll dig into that and fix it later.

**Comment by algon-320** [comment]
I fixed it in 6719db022a50020b31e4002a76e89eec274dfdb6. Thank you again @klamonte !
